Python/Product/PythonTools/PythonTools/Intellisense/AnalysisEntry.cs

```diff
@@ -51,7 +51,7 @@ internal sealed class AnalysisEntry : IDisposable {
         ) {
             Analyzer = analyzer;
             Path = path;
-            DocumentUri = documentUri;
+            DocumentUri = documentUri ?? (!string.IsNullOrEmpty(path) ? new Uri(path) : null);
```

Comment:
Since DocumentUri is the canonical name, I'd rather not infer it from another property (or else how are we going to be able to match it later). How about we throw if it's null?

Reply:
OK. However, `ProjectEntry` does derive it from the path via `MakeDocumentUri`.

Reply:
@MikhailArkhipov Only for legacy reasons. None of the language server related code ever passes `null` (though some of it does convert URI back to file system...), but some of our tests and the legacy DB system don't/won't use document URIs at all.